vm/systemSmartContracts/governance.go

```diff
@@ -147,73 +165,268 @@ func (g *governanceContract) init(args *vmcommon.ContractCallInput) vmcommon.Ret
 	return vmcommon.Ok
 }
 
-func (g *governanceContract) changeConfig(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	if !bytes.Equal(g.ownerAddress, args.CallerAddr) {
-		g.eei.AddReturnMessage("changeConfig can be called only by owner")
+// proposal creates a new proposal from passed arguments
+func (g *governanceContract) proposal(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if args.CallValue.Cmp(g.baseProposalCost) != 0 {
+		g.eei.AddReturnMessage("invalid proposal cost, expected " + g.baseProposalCost.String())
+		return vmcommon.OutOfFunds
+	}
+	err := g.eei.UseGas(g.gasCost.MetaChainSystemSCsCost.Proposal)
+	if err != nil {
+		g.eei.AddReturnMessage("not enough gas")
+		return vmcommon.OutOfGas
+	}
+	if len(args.Arguments) != 3 {
+		g.eei.AddReturnMessage("invalid number of arguments, expected 3")
+		return vmcommon.FunctionWrongSignature
+	}
+	if !g.isWhiteListed(args.CallerAddr) {
+		g.eei.AddReturnMessage("called address is not whiteListed")
+		return vmcommon.UserError
+	}
+	gitHubCommit := args.Arguments[0]
+	if len(gitHubCommit) != githubCommitLength {
```

Comment:
take into consideration that the githubCommitLength is 40 hex characters but 20 bytes long; just to take into account that he arguments[0] is a byte[] and what encoding is there

Reply:
arguments are without encoding. encoding is needed when creating the transaction data- but they arrive in the vm hex.decoded

Reply:
ok